### ISSUE TITLE
[Feature request] Hide restricted activities from progress bar 

### ISSUE THREAD

**Issue by izendegi** [jonof/moodle-block_completion_progress#72]
Hi,

We've realized that the progress bar doesn't show hidden activities, but those activities with access restrictions are always shown even if they're not available, and that could confuse students.

It would be great if the progress bar could only show the activities each student could access, and therefore the percentage shown in the teachers' report would be more accurate too.

**Comment by deraadt** [comment]
Unless something has changed, students should only see activities in the
bar that they can see in the course and the percentage, if it is shown, is
based on that. Even on the Overview page, this is checked for each student.

Michael

**Comment by izendegi** [comment]
Hi @deraadt 

I've been diving into this and you're right, the activities shown on the completion progress bar are those **visible** for the students, even if they are not **accessible** for them.

So, if an activity has some access restriction but it's not fully hidden for those who haven't got access (it has the access restriction rule's eye opened), it appears on every students' bar even if some of them can't actually access it.

https://docs.moodle.org/310/en/Restrict_access_settings#Hiding_the_conditions

I guess this behaviour makes sense for some kind of restrictions (by date, by activity completion...), but it may be a nice improvement to allow the teachers to choose between showing in the progress bar "all the activities the student can see, even if they are not currently accessible" (current behaviour) and "only the activities the student can access".

**Comment by mhughes2k** [comment]
I think I've found a variation on this, but activities that are in a section that has restricted access conditions to prevent a student seeing them seem to be getting displayed. I'm trying to get a clearer definition but thought would lodge this here for info.

**Comment by mhughes2k** [comment]
Nope, I can't actually reproduce this, but as far as I can tell the restrict access condition to prevent the student's from seeing the activity does *seem* to be working.

If I set up an course with 2 sections, one with a RA to require membership of a group "A", and then 3 page activities: 1 with no RA (A1), 1 in the RA restricted section (A2) and the last one with the RA to require membership of group "B" (A3) (All RAs set with the eye-crossed out to completly hide the activity or section).

If I assign 2 users to each groups separately, then on an instance of the completion progress bar that is set to "All activities", the student overview displays 2 activities for each user.

User 1 who's a member of Group 1 gets A1 & A2
User 2 who's a member of Group 2 gets A1 & A3
